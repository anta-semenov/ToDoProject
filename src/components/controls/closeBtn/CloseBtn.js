import React from 'react'
import './CloseBtn.less'

export default class CloseBtn extends React.Component {
  render() {
    return (
      <div className={`close close--${this.props.appearance}`} title='Close' onClick={() => this.props.onClick()}>
        <div className='close__sign'/>
      </div>
    )
  }
}

CloseBtn.propTypes = {
  appearance: React.PropTypes.oneOf(['task-info', 'default']),
  onClick: React.PropTypes.func.isRequired
}

CloseBtn.defaultProps = {
  appearance: 'default'
}
