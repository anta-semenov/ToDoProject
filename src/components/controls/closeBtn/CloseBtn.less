@import '../../../styles/variables.less';

.close {
  &::before {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    border-radius: 50%;
    background-color: @color-bg-light-gray;
    opacity: 0;
    transform: scale(0);
    transition: @transition-basic;
  }
  &:hover::before {
    opacity: 1;
    transform: scale(1);
  }
  &.close--task-info {
    position: absolute;
    top: 5px;
    right: 5px;
    width: 30px;
    height: 30px;
  }
}

.close__sign {
  position: relative;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  transition: @transition-basic;
  cursor: pointer;

  &::before,
  &::after {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    width: 1px;
    background-color: @color-font-light;
    transition: @transition-basic;

    .close--task-info & {
      height: 16px;
    }
  }
  &::before {
    transform: translate(-50%, -50%) rotate(45deg);
  }
  &::after {
    transform: translate(-50%, -50%) rotate(-45deg);
  }
  .close:hover & {
    &::before,
    &::after {
      background-color: @color-font-medium;
    }
  }
  .task-info.is-open .close--task-info & {
    animation: close-in 0.6s ease-out 0.1s;
    animation-fill-mode: backwards;
  }
}

@keyframes close-in {
  0% {transform: scale(0); opacity: 0;}
  100% {transform: scale(1); opacity: 1;}
}
