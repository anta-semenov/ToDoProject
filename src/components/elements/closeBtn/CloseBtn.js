import PropTypes from 'prop-types';
import React from 'react'
import './CloseBtn.less'

const CloseBtn = ({ appearance = 'default', onClick }) => (
  <div className={`close close--${appearance}`} title='Close' onClick={() => onClick()}>
    <div className='close__sign'/>
  </div>
)

CloseBtn.propTypes = {
  appearance: PropTypes.oneOf(['task-info', 'default']),
  onClick: PropTypes.func.isRequired
}

export default CloseBtn
