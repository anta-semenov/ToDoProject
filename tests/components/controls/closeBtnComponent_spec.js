import React from 'react'
import { shallow } from 'enzyme'
import CloseBtn from '../../../src/components/elements/closeBtn/CloseBtn'

describe('CloseBtn component', () => {
  test('Should render component with right class for empty appearance prop', () => {
    const closeBtn = shallow(<CloseBtn onClick={jest.fn()} />)
    expect(closeBtn).toMatchSnapshot()
  })
  test('Should render right class for particular appearance prop', () => {
    const closeBtn = shallow(<CloseBtn appearance="task-info" onClick={jest.fn()} />)
    expect(closeBtn).toMatchSnapshot()
  })
})
