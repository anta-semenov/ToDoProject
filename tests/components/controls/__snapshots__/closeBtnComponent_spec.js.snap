// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseBtn component Should render component with right class for empty appearance prop 1`] = `
<div
  className="close close--default"
  onClick={[Function]}
  title="Close"
>
  <div
    className="close__sign"
  />
</div>
`;

exports[`CloseBtn component Should render right class for particular appearance prop 1`] = `
<div
  className="close close--task-info"
  onClick={[Function]}
  title="Close"
>
  <div
    className="close__sign"
  />
</div>
`;
